docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+
+[React](https://reactjs.org/) is used for more complex parts of Wagtail, such as the sidebar, commenting system, and the Draftail rich-text editor.
+For basic JavaScript-driven interaction, Wagtail is migrating towards [Stimulus](https://stimulus.hotwired.dev/).
+
+You don't need to know or use these libraries to add your custom behaviour to elements, and in many cases vanilla (plain) JavaScript will work fine, but Stimulus is the recommended approach for anything that vanilla JavaScript can't do.
+
+You don't need to have Node.js tooling running for your custom Wagtail installation for many customisations built on these libraries, but in some cases, such as building packages, it may make more complex development easier.
+
+```{note}
+it's recommended that you avoid using jQuery as this will be removed in a future version of Wagtail.
+```
+
+(extending_client_side_injecting_javascript)=
+
+## Adding custom JavaScript
+
+Within Wagtail's admin interface, there are a few ways to add JavaScript.
+
+The simplest way is to add global JavaScript files via hooks, see [](insert_editor_js) and [](insert_global_admin_js).
+
+For JavaScript added when a specific Widget is used you can add an inner `Media` class to ensure that the file is loaded when the widget is used, see [Django's docs on their form `Media` class](https://docs.djangoproject.com/en/stable/topics/forms/media/#assets-as-a-static-definition).
+
+In a similar way, Wagtail's [template components](template_components) provide a `media` property or `Media` class to add scripts when rendered.
+
+These will ensure the added files are used in the admin after the core JavaScript admin files are already loaded.
+
+(extending_client_side_using_events)=
+
+## Extending with DOM events
+
+When approaching client-side customisations or adopting new components, try to keep the implementation simple first, you may not need any knowledge of Stimulus, React, JavaScript Modules or a build system to achieve your goals.
+
+The simplest way to attach behaviour to the browser is via DOM Events and plain (vanilla) JavaScript.
+
+For example, if you want to attach some logic to a field value change in Wagtail you can add an event listener, check if it's the correct element and change what you need.
+
+```javascript
+// myapp/static/js/events.js
+document.addEventListener('change', function (event) {
+    if (event.target) {
+        console.log('field has changed', event.target);
+    }
+});
+```
+
+Alternatively, you can write JavaScript logic to achieve this by listening to all click events and checking if the clicked element corresponds to the comments section.
+
+```javascript
+// myapp/static/js/events.js
+document.addEventListener('click', function (event) {
+    if (!event.target) return;
+
+    isCommentsToggle = event.target.matches(
+        '[data-side-panel-toggle="comments"]',
+    );
```

Comment:
I’m finding this problematic for the above reason (executing JS on every click seems wasteful / dated), and also because we’re now document those `data-` attributes as things you can rely on in your customisation. I’d prefer we stuck to documented parts of Wagtail, or at least examples that are less tied to our implementation details.

If we keep this,

```suggestion
document.addEventListener('click', function (event) {
    if (!event.target) return;

    const isCommentsToggle = event.target.matches(
        '[data-side-panel-toggle="comments"]',
    );
```

Reply:
I'll try to think of a better example here. I agree we probably don't want to highlight this data attribute specifically. 

However, as above, I'd push back on this being dated. Remember the target audience is people trying to solve things they cannot solve with html / Python alone.

Reply:
As per comment above, will remove this example, keep it generic and just focus on the goal of reminding devs that Events exist.